### PR TITLE
Fix content treated as plain text after expression in some cases

### PR DESCRIPTION
**Pull request by MoustaphaDev**
## Changes

Second part of the fix for #609
Fixes an issue where when there are nested expressions, subsequent content was incorrectly treated as plain text in some cases.

## Testing

Added a printer test and fixed an incorrect test

## Docs

N/A bug fix

### CODE REVIEW

File: internal/printer/printer_test.go [hunk 1]
```diff
@@ -2848,7 +2855,7 @@ const items = ["Dog", "Cat", "Platipus"];
...
+				code: `${($$render` + BACKTICK + `${$$renderComponent($$result,'Fragment',Fragment,{},{"default": () => $$render` + BACKTICK + `${$$renderComponent($$result,'Fragment',Fragment,{},{"default": () => $$render` + BACKTICK + `${$$unescapeHTML(` + BACKTICK + `<${Node.tag} ${stringifyAttributes(Node.attributes)}>` + BACKTICK + `)}` + BACKTICK + `,})}${Node.children.map((child) => ($$render` + BACKTICK + `${$$renderComponent($$result,'Astro.self',Astro.self,{"node":(child)})}` + BACKTICK + `))}${$$renderComponent($$result,'Fragment',Fragment,{},{"default": () => $$render` + BACKTICK + `${$$unescapeHTML(` + BACKTICK + `</${Node.tag}>` + BACKTICK + `)}` + BACKTICK + `,})}` + BACKTICK + `,})}` + BACKTICK + `)}`,
```
Comment (MoustaphaDev) [inline internal/printer/printer_test.go:2858]: Here the backtick was closed after the closing parenthesis which is incorrect. You can see that in [the right pane of this visualization](https://live-astro-compiler.vercel.app/?editor-state=WzMxLDEzOSw4LDAsMjI4LDEwNiw5NSwxMDEsMiwzLDEwOSwxNDMsMjA1LDEwNiwxOTUsNDgsMTYsMTMyLDk1LDY5LDEzNiwyOCwxMDgsOCwyMDYsMjIxLDIxNiwxMzQsMTYsOCwyMCw3NCw0MSw5LDE2NSwyMTUsMTA4LDE3MywxODEsMzUsMjA4LDE0MywxNDUsMjE0LDE0OCw4NiwyMzIsMjIxLDQzLDIxOSw4NCwyMzcsMzMsMTgzLDIyMSwxNTcsMTExLDcwLDE2MywxOTIsMTIzLDQzLDE0NCwyMTUsNjAsMjAsMjA1LDIxNywxOTMsMTY4LDIwOSw4MCwxNTEsMzksMjMwLDE0NSwyMzQsNTksMTA1LDIxMywxMzQsOTEsMTc5LDExLDQ3LDksMTczLDgsMTk4LDIwMCwxMTgsMTkzLDE0NywxNDcsMTAyLDE0OCwxOTUsMjE1LDE0NSwyMTAsMjQ0LDQ5LDE5LDI1MCw5OCwyMTMsMzMsMjM5LDEwMSwyMzYsMTEwLDE0NSwyOSwxODYsMjA1LDIxNSwyMjMsMTY1LDE4LDE0LDc3LDE2NSw5Nyw0MiwxMzgsMTE3LDQzLDg5LDIxOSwxNzcsMTYyLDU3LDE2Niw0NCw5MSwxMjEsODQsMyw1MSwxMzcsMTA4LDE5NSw0Miw0NiwyMDYsMTc4LDE0MCwxNDMsMTg3LDI4LDI1NCwxNDksMjE3LDMwLDEwNSwxNCwxODUsMTI2LDI1LDI0OSwxNTgsMTI3LDkwLDM5LDIyMiwyOSw3NiwxOSwxMCw5NCwxNSwxNjAsNjAsMjM4LDE4NSwyMjIsMTkwLDc0LDE0LDE0MCwzMSwxNzIsMjExLDksMTU1LDE5MiwxMjEsMTI0LDE4MSw5NCwxNDYsMTgwLDEzNCwyMTUsMjI4LDIzMCwxOTYsMTAxLDIyNCwyMDEsMTYsNTgsMywyMzQsMjM3LDI0MiwxNTYsMTI0LDksMzEsMTY0LDY2LDMsMTIyLDczLDU3LDEyOSwxOSwyMSw0NCwyMTMsMjExLDIyMSwzNiwyNCwxNDgsMjUyLDcwLDExMywyNTQsMzUsMjEwLDYxLDM5LDE1NywxNzIsMTU4LDE2MCwxNjcsOTIsMzcsMTEsMjMsMjQ0LDE3OSwxNjIsMTA3LDExMSw4MywyMDksMTcxLDE3OCwxOTEsNjgsMjUyLDEsMTEzLDIyMCwxMDIsMjcsMTUzLDEsMCwwXQ%3D%3D).

File: .changeset/mighty-olives-wash.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/compiler': patch
+---
+
+Fix an issue where when there are nested expressions, subsequent content was incorrectly treated as plain text in some cases.
```
Comment (sarah11918) [inline .changeset/mighty-olives-wash.md:5]:
```suggestion
Fixes an issue where when there are nested expressions, subsequent content was incorrectly treated as plain text in some cases.
```

**Review by sarah11918 — COMMENTED** [review]
Great message, just standardizing the verb tense! 🥳

File: .changeset/mighty-olives-wash.md (hunk 2 above)
Comment (MoustaphaDev) [reply .changeset/mighty-olives-wash.md]: Noted for the future, thanks @sarah11918 🙌

**Review by natemoo-re — APPROVED** [review]
This looks totally reasonable, thank you!

**Pull request merged**